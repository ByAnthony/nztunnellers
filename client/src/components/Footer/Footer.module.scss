@use 'src/index.scss';

.footer {
  margin-top: index.$l-margin;
  padding-bottom: index.$base-margin;

  @include index.component-margin;

  @include index.desktop {
    margin: 0 auto;
    padding: index.$base-margin;
    max-width: 57rem;
  }

  p {
    font-size: 1.2rem;
    font-weight: 500;
    text-align: center;
  }

  .map {
    padding-top: index.$m-margin;
    padding-bottom: index.$base-margin;
    display: flex;
    justify-content: space-between;
    align-items: center;
    border-bottom: 1px solid index.$primary-light-color;

    @include index.tablet {
      padding-top: index.$m-margin;
      padding-bottom: index.$m-margin;
    }

    @include index.desktop {
      padding-top: index.$base-margin;
    }

    .links {
      display: inline-flex;
      flex-direction: column;
      gap: index.$s-margin;

      @include index.tablet {
        flex-direction: row;
        gap: index.$base-margin;
      }

      @include index.desktop {
        flex-direction: row;
        gap: index.$l-margin;
      }
    }

    .map-link a {
      padding-bottom: 4px;
      font-size: 1rem;
      font-weight: 500;
      color: index.$text-color;
      border-bottom: 1px solid index.$secondary-color;
      transition: color 0.9s ease-in-out, color 0.9s ease-in-out,
        border-bottom 0.9s ease-in-out, border-bottom 0.9s ease-in-out;

      &:hover {
        padding-bottom: 4px;
        border-bottom: 1px solid index.$secondary-color;
        color: index.$secondary-color;
      }
    }
  }

  button {
    padding: 8px 18px 6px 18px;
    font-family: index.$main-font;
    font-size: 2.25rem;
    font-weight: 200;
    text-align: center;
    color: index.$secondary-color;
    border: 1px solid index.$primary-light-color;
    border-radius: index.$border-radius;
    background-color: index.$primary-light-color;
    transition: border-bottom 0.9s ease-in-out, border-bottom 0.9s ease-in-out,
      border 0.9s ease-in-out, border 0.9s ease-in-out,
      background-color 0.9s ease-in-out, background-color 0.9s ease-in-out;
    cursor: pointer;

    &:hover {
      border: 1px solid index.$secondary-color;
      background-color: index.$primary-medium-color;
    }
  }

  .support {
    padding-top: index.$base-margin;
    display: flex;
    justify-content: space-evenly;
    align-items: center;

    @include index.tablet {
      justify-content: center;
      gap: index.$l-margin;
    }

    @include index.desktop {
      padding-top: index.$base-margin;
      justify-content: center;
      gap: index.$xl-margin;
    }

    img {
      width: 100px;
      text-align: center;

      @include index.tablet {
        width: 125px;
      }

      @include index.desktop {
        width: 125px;
      }
    }
  }
}
