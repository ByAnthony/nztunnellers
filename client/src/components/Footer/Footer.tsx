import STYLES from './Footer.module.scss';

export function Footer() {
  const handleClick = () => {
    window.scrollTo(0, 0);
  };

  return (
    <div className={STYLES.footer}>
      <div className={STYLES.support}>
        <p>Support</p>
        <div className={STYLES['support-logos']}>
          <a href="https://www.univ-artois.fr/artois-university" aria-label="Go to The Artois University website."><img src="../images/support/logo-univ-artois-blanc_0.png" alt="" /></a>
          <a href="https://www.irsem.fr/en/" aria-label="Go to The Institute for Strategic Research website."><img src="../images/support/irsem-white.png" alt="" /></a>
        </div>
      </div>
      <div className={STYLES.links}>
        <button type="button" className={STYLES['scroll-top']} onClick={handleClick}>&uarr;</button>
      </div>
    </div>
  );
}
