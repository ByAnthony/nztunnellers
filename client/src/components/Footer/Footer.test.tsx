import '@testing-library/jest-dom';
import {
  fireEvent, render, screen,
} from '@testing-library/react';

import { Footer } from './Footer';

const component = <Footer />;

test('renders the component correctly', () => {
  const { asFragment } = render(component);

  expect(asFragment()).toMatchSnapshot();
});

test('can scroll to top of the page', () => {
  global.scrollTo = jest.fn();

  render(component);
  fireEvent.click(screen.getByRole('button'));

  expect(global.scrollTo).toHaveBeenCalledWith(0, 0);
});
