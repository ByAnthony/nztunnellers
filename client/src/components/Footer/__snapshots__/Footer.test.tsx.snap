// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the component correctly 1`] = `
<DocumentFragment>
  <div
    class="footer"
  >
    <div
      class="map"
    >
      <div
        class="links"
      >
        <div
          class="link"
        >
          <a
            href="/history"
          >
            History
          </a>
        </div>
        <div
          class="link"
        >
          <a
            href="/tunnellers"
          >
            Tunnellers
          </a>
        </div>
        <div
          class="link"
        >
          <a
            href="/about-us"
          >
            About Us
          </a>
        </div>
      </div>
      <button
        aria-label="Go back to the top of the page"
        class="scroll-top"
        type="button"
      >
        â†‘
      </button>
    </div>
    <div
      class="support"
    >
      <div>
        <a
          aria-label="Go to The Artois University website"
          href="https://www.univ-artois.fr/artois-university"
        >
          <img
            alt="Artois University homepage"
            src="/images/support/logo-univ-artois-blanc_0.png"
          />
        </a>
      </div>
      <div>
        <a
          aria-label="Go to The Institute for Strategic Research website"
          href="https://www.irsem.fr/en/"
        >
          <img
            alt="Institute for Strategic Research homepage"
            src="/images/support/irsem-white.png"
          />
        </a>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
